extensions/amp-sidebar/1.0/sidebar-animations-hook.js

```diff
@@ -0,0 +1,192 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Side, useValueRef} from './sidebar-config';
+import {assertDoesNotContainDisplay, setStyles} from '../../../src/style';
+import {useLayoutEffect, useRef} from '../../../src/preact';
+
+const ANIMATION_DURATION = 350;
+const ANIMATION_EASE_IN = 'cubic-bezier(0,0,.21,1)';
+
+const ANIMATION_KEYFRAMES_FADE_IN = [{'opacity': '0'}, {'opacity': '1'}];
+const ANIMATION_KEYFRAMES_SLIDE_IN_LEFT = [
+  {'transform': 'translateX(-100%)'},
+  {'transform': 'translateX(0)'},
+];
+const ANIMATION_KEYFRAMES_SLIDE_IN_RIGHT = [
+  {'transform': 'translateX(100%)'},
+  {'transform': 'translateX(0)'},
+];
+
+const ANIMATION_STYLES_SIDEBAR_LEFT_INIT = {'transform': 'translateX(-100%)'};
+const ANIMATION_STYLES_SIDEBAR_RIGHT_INIT = {'transform': 'translateX(100%)'};
+const ANIMATION_STYLES_BACKDROP_INIT = {'opacity': '0'};
+const ANIMATION_STYLES_SIDEBAR_FINAL = {'transform': ''};
+const ANIMATION_STYLES_BACKDROP_FINAL = {'opacity': ''};
+
+/** @private @enum {number} */
+const Direction = {
```

Comment:
bikeshed: `AnimationState`? With `OPENING`, `CLOSING`, and `NONE`? I worry about `Direction` being too overloaded with meaning.

Reply:
Thanks, per offline conversation updated to `AnimationState` and kept `OPENING`, `CLOSING`, and `NOT_ANIMATING`.  Open to additional conversation if needed :)

Reply:
Simplified this to just a `true` `false` state and removed the enum.  We are able to get to this effect only when the animation changes direction so we do not need to track the direction, just whether or not the animation is currently running.